src/components/configurable-columns/configurable-columns.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import I18n from 'i18n-js';
+import tagComponent from '../../utils/helpers/tags';
+import { DraggableContext } from './../drag-and-drop';
+import Button from './../button';
+import ConfigurableColumnRow from './configurable-column-row';
+import Form from './../form';
+
+class ConfigurableColumns extends React.Component {
+  static propTypes = {
+    /**
+     * Children elements.
+     *
+     * @property children
+     * @type {Node}
+     */
+    children: PropTypes.node,
+
+    /**
+     * A custom class name for the component.
+     *
+     * @property className
+     * @type {String}
+     */
+    className: PropTypes.string,
+
+    /**
+     * Callback triggered when the form is canceled.
+     *
+     * @property onCancel
+     * @type {Function}
+     */
+    onCancel: PropTypes.func.isRequired,
+
+    /**
+     * Callback triggered when an item is dragged.
+     *
+     * @property onDrag
+     * @type {Function}
+     */
+    onDrag: PropTypes.func.isRequired,
+
+    /**
+     * Callback triggered when when the reset button is pressed.
+     *
+     * @property onReset
+     * @type {Function}
+     */
+    onReset: PropTypes.func,
+
+    /**
+     * Callback triggered when the form is saved.
+     *
+     * @property onSave
+     * @type {Function}
+     */
+    onSave: PropTypes.func.isRequired
+  }
+
+  onReset = (event) => {
+    event.preventDefault();
+    this.props.onReset();
```

Comment:
Could wrap a if statement around this call as its not a required prop. Don't think it will complain though?